octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/EntityModelEditor.java

```diff
@@ -148,45 +150,48 @@ private void setIsDirty(boolean isDirty) {
     }
 
     @Override
-    public void setFocus() {
-    }
+    public void setFocus() {}
 
     @Override
     public void doSave(IProgressMonitor monitor) {
         UpdateEntityJob updateEntityJob = new UpdateEntityJob("Saving " + entityModelWrapper.getEntityType(), entityModelWrapper.getEntityModel());
-
-        updateEntityJob.addJobChangeListener(new JobChangeAdapter() {
-            @Override
-            public void done(IJobChangeEvent event) {
-                OctaneException octaneException = updateEntityJob.getOctaneException();
-
-                if (octaneException == null) {
+        updateEntityJob.schedule();
+        
+        try {
+            updateEntityJob.join(1000 * 10, monitor);
+
+            OctaneException octaneException = updateEntityJob.getOctaneException();
+            if (octaneException == null) {
+                loadEntity();
+                
+            } else {
+                EntityDetailErrorDialog errorDialog = new EntityDetailErrorDialog(rootComposite.getShell());
+                errorDialog.addButton("Back", () -> errorDialog.close());
+                errorDialog.addButton("Refresh", () -> {
                     loadEntity();
-
-                } else {
-                    Display.getDefault().asyncExec(() -> {
-
-                        EntityDetailErrorDialog errorDialog = new EntityDetailErrorDialog(rootComposite.getShell());
-
-                        errorDialog.addButton("Back", () -> errorDialog.close());
-
-                        errorDialog.addButton("Refresh", () -> {
-                            loadEntity();
-                            errorDialog.close();
-                        });
-                        errorDialog.addButton("Open in browser", () -> {
-                            entityService.openInBrowser(entityModelWrapper.getReadOnlyEntityModel());
-                            errorDialog.close();
-                        });
-
-                        errorDialog.open(octaneException, "Saving entity failed");
-
-                    });
-                }
+                    errorDialog.close();
+                });
+                errorDialog.addButton("Open in browser", () -> {
+                    entityService.openInBrowser(entityModelWrapper.getReadOnlyEntityModel());
+                    errorDialog.close();
+                });
+                
+                errorDialog.open(octaneException, "Saving entity failed");
+                
+                //This will stop the window from closing if an error occurs 
+                //when saving from the dialog that appears when you close a dirty tab
+                monitor.setCanceled(true); 

```

Comment:
@atothhpe Looking at the code, it seems that doSave is only called twice and with null as a param.
monitor.setCanceled(true); will definitely throw an exception.
What will happen to "updateEntityJob.join(1000 * 10, monitor);" when monitor is null?